server/stream.go

```diff
@@ -1765,16 +1765,17 @@ func (mset *stream) setSourceConsumer(iname string, seq uint64) {
 func (mset *stream) processSourceMsgs(si *sourceInfo) {
 	s := mset.srv
 	defer s.grWG.Done()
+
+	if si == nil {
```

Comment:
I am assuming this is not related to this issue, but required otherwise could crash during the defer.

Reply:
yes correct, was a pending change I was going to make so just put it in here.

Want to make sure tests are green so will keep watching and merge after.